src/core.c/allomorphs.pm6

```diff
@@ -1,11 +1,39 @@
+my class Allomorph is Str {
+    multi method Bool(::?CLASS:D:) { self.Numeric.Bool }
+
+    method succ(Allomorph:D:) { self.Numeric.succ }
+    method pred(Allomorph:D:) { self.Numeric.pred }
+
+    multi method Str(Allomorph:D:) {
+        nqp::getattr_s(self,Str,'$!value')
+    }
+
+    multi method WHICH(Allomorph:D:) {
+        nqp::box_s(
+          nqp::join('|',nqp::list_s(
+            self.^name,
+            self.Numeric.WHICH,
+            self.Str.WHICH
```

Comment:
Is there a case where the type and `Str` value would be the same, but `self.Numeric.WHICH` would be different? I struggle to imagine one.

Reply:
Only if you artificially construct allomorphs.  Which people can do, and therefore will.

Reply:
Goodness, why... :-) But yes, I guess it doesn't hurt too much to be robust against it.